opendevin/server/agent/agent.py

```diff
@@ -138,6 +138,10 @@ async def on_event(self, event: Event):
             await self.send(event.to_dict())
         return
 
-    def close(self):
+    async def close(self):
+        """Cleanly exits an AgentUnit.
+
+        Execution awaited by the AgentManager.
+        """
         if self.controller is not None:
-            self.controller.close()
```

Comment:
This change looks make sense. But I thought if we try to call async function in a normal function, our lint or python interpreter wil throw warning during running. Why we didn't find this problem before? 
For example, when I try this simple demo, the interpreter will throw warning. cc @li-boxuan  Do you have any idea?
<img width="837" alt="image" src="https://github.com/OpenDevin/OpenDevin/assets/33971064/6386085f-640f-4211-ad93-8723d60b2fdb">

Reply:
@yufansong does calling `hello()` print hello world? I thought the coroutine won't execute unless you await it.

Reply:
Calling `hello` not print, so the interpreter will report the warning. I just try to reproduce the logic in this case which also not use `await`.

Reply:
> I'm not sure, but I had seen that warning when I shutdown the application with control-c.

If you see the warning, that make sense!